harper-core/src/linting/ask_no_preposition.rs

```diff
@@ -0,0 +1,196 @@
+use crate::{
+    Span, Token,
+    linting::{Lint, LintKind, PatternLinter, Suggestion},
+    patterns::{Pattern, SequencePattern, WordSet},
+};
+
+pub struct AskNoPreposition {
+    pattern: Box<dyn Pattern>,
+}
+
+impl Default for AskNoPreposition {
+    fn default() -> Self {
+        let verbs = WordSet::new(&[
+            "ask", "asks", "asked", "asking", "tell", "tells", "told", "telling",
+        ]);
+
+        let objs = WordSet::new(&["me", "you", "him", "her", "it", "us", "them", "one"]);
+
+        let pattern = SequencePattern::default()
+            .then(verbs)
+            .then_whitespace()
+            .then_exact_word("to")
+            .then_whitespace()
+            .then(objs);
+
+        Self {
+            pattern: Box::new(pattern),
+        }
+    }
+}
+
+impl PatternLinter for AskNoPreposition {
+    fn pattern(&self) -> &dyn Pattern {
+        self.pattern.as_ref()
+    }
+
+    fn match_to_lint(&self, toks: &[Token], src: &[char]) -> Option<Lint> {
+        if toks.len() < 5 {
+            return None;
+        }
+
+        let verb = toks[0].span.get_content_string(src).to_lowercase();
+        let span = Span::new(toks[2].span.start, toks[3].span.end);
+
+        Some(Lint {
+            span,
+            lint_kind: LintKind::WordChoice,
+            suggestions: vec![Suggestion::ReplaceWith(Vec::new())],
+            message: format!(
+                "The verb `to {verb} someone` should not be preceded by the preposition “to”."
```

Comment:
The verb itself can be preceded by "to" but the object of the verb, in this case the pronoun, shouldn't be preceded by "to". Or you could say the verb shouldn't be followed by "to".